modules/fedimint-wallet/src/lib.rs

```diff
@@ -1687,24 +1789,31 @@ pub async fn run_broadcast_pending_tx(db: Database, rpc: DynBitcoindRpc, tg_hand
 }
 
 pub async fn broadcast_pending_tx(mut dbtx: DatabaseTransaction<'_>, rpc: &DynBitcoindRpc) {
-    let pending_tx = dbtx
+    let pending_tx: Vec<PendingTransaction> = dbtx
         .find_by_prefix(&PendingTransactionPrefixKey)
         .await
         .map(|res| {
-            let (key, val) = res.expect("DB Error");
-            (key, val)
+            let (_key, val) = res.expect("DB Error");
+            val
         })
-        .collect::<Vec<(_, _)>>()
+        .collect::<Vec<_>>()
         .await;
-
-    for (_, PendingTransaction { tx, .. }) in pending_tx {
-        debug!(
-            tx = %tx.txid(),
-            weight = tx.weight(),
-            "Broadcasting peg-out",
-        );
-        trace!(transaction = ?tx);
-        let _ = rpc.submit_transaction(tx).await;
+    let rbf_txids: BTreeSet<Txid> = pending_tx
+        .iter()
+        .filter_map(|tx| tx.rbf.clone().map(|rbf| rbf.txid))
+        .collect();
+
+    for PendingTransaction { tx, .. } in pending_tx {
+        if !rbf_txids.contains(&tx.txid()) {
+            debug!(
+                tx = %tx.txid(),
+                weight = tx.weight(),
+                output = ?tx.output,
+                "Broadcasting peg-out",
+            );
+            trace!(transaction = ?tx);
+            let _ = rpc.submit_transaction(tx).await;
+        }
```

Comment:
This should be perfomant enough for now, eventually we could split the RBFed and non-RBFed txns into different DB "tables".